src/main/java/org/jenkinsci/test/acceptance/plugins/gradle/GradleStep.java

```diff
@@ -23,26 +23,84 @@
  */
 package org.jenkinsci.test.acceptance.plugins.gradle;
 
+import hudson.util.VersionNumber;
 import org.jenkinsci.test.acceptance.po.*;
 
 @Describable("Invoke Gradle script")
 public class GradleStep extends AbstractStep implements BuildStep {
-    public final Control name = control("");
-    public final Control file = control("buildFile");
-    public final Control dir = control("rootBuildScriptDir");
-    public final Control switches = control("switches");
-    public final Control tasks = control("tasks");
+
+    private final Control file = control("buildFile");
+    private final Control dir = control("rootBuildScriptDir");
+    private final Control switches = control("switches");
+    private final Control tasks = control("tasks");
+    private final Control useWrapper = control("useWrapper[true]");
+    private final Control makeWrapperExecutable = control("makeExecutable");
+    private final Control passAllAsProjectProperties = control("passAllAsProjectProperties");
+    private final Control passAllAsSystemProperties = control("passAllAsSystemProperties");
+    private final Control forceGradleHomeToUseWorkspace = control("useWorkspaceAsHome");
 
     public GradleStep(Job parent, String path) {
         super(parent, path);
     }
 
-    public void useVersion(String version) {
+    private void ensureAdvancedOptionsOpen() {
+        if(control("advanced-button").exists()) {
```

Comment:
Extract control("advanced-button") into local variable so it is called only once.